### PR TITLE
RESTWS-834: Add REST resource for Database Changes

### PR DESCRIPTION
**Pull request by BartlomiejRasztabiga**
## Description of what I changed
Added DatabaseChangeResource2_0 that allows listing all OpenMRSChangeSets and listing one by its id.

## Issue I worked on
see https://issues.openmrs.org/browse/RESTWS-834

PR with documentation: https://github.com/openmrs/openmrs-contrib-rest-api-docs/pull/154

## Checklist: I completed these to help reviewers :)
- [X] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [X] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/DatabaseChangeResource2_0.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+		return null;
+	}
+
+	private List<DatabaseUpdater.OpenMRSChangeSet> getDatabaseChanges() {
+		List<DatabaseUpdater.OpenMRSChangeSet> databaseChanges;
+		try {
+			databaseChanges = DatabaseUpdater.getDatabaseChanges();
+		}
+		catch (Exception e) {
+			e.printStackTrace();
```
Comment (dkayiwa) [inline omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/DatabaseChangeResource2_0.java:156]: https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotprintout

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		return "databasechange";
+	}
+
+	@Override
+	public String getUuid() {
+		try {
+			return DatabaseUpdater.getDatabaseChanges().get(0).getId();
+		}
+		catch (Exception e) {
+			return null;
```
Comment (dkayiwa) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java:28]: What kind of exception is this?

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		}
+	}
+
+	@Override
+	public long getAllCount() {
+		try {
+			return Math.min(DatabaseUpdater.getDatabaseChanges().size(), 50); // page is max 50
+		}
+		catch (Exception e) {
+			return 0;
```
Comment (dkayiwa) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java:38]: Same as above.

File: omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/DatabaseChangeResource2_0.java (hunk 1 above)
Comment (BartlomiejRasztabiga) [reply omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/DatabaseChangeResource2_0.java]: https://github.com/openmrs/openmrs-module-webservices.rest/pull/491/commits/599a1179888fc0ab333a59d294aa96bb2ec7ab04

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java (hunk 2 above)
Comment (BartlomiejRasztabiga) [reply omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java:28]: `DatabaseUpdater.getDatabaseChanges()` can throw `Exception`. The real exception is probably more specific but that's how it's coded in the core.

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java (hunk 3 above)
Comment (BartlomiejRasztabiga) [reply omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java:38]: https://github.com/openmrs/openmrs-module-webservices.rest/pull/491#discussion_r666344449

File: omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/DatabaseChangeResource2_0.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	}
+
+	@Override
+	public DatabaseUpdater.OpenMRSChangeSet newDelegate() {
+		return null;
+	}
+
+	private List<DatabaseUpdater.OpenMRSChangeSet> getDatabaseChanges() {
+		List<DatabaseUpdater.OpenMRSChangeSet> databaseChanges;
+		try {
```
Comment (dkayiwa) [inline omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/DatabaseChangeResource2_0.java:153]: What happens if we do not catch this exception?

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java (hunk 2 above)
Comment (dkayiwa) [reply omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java:28]: What happens if we do not catch this exception?

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java (hunk 3 above)
Comment (dkayiwa) [reply omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java:38]: What happens if we do not catch this exception?

File: omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/DatabaseChangeResource2_0.java (hunk 4 above)
Comment (BartlomiejRasztabiga) [reply omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/DatabaseChangeResource2_0.java:153]: It's a checked exception so either we wrap this call in try-catch block or add it to the method signature, that this method throws. If we add that to the method signature, then we would have to add the same `throws Exception` to `getByUniqueId` and `doGetAll`, which is illegal, as they don't throw anything in the interface.

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java (hunk 3 above)
Comment (BartlomiejRasztabiga) [reply omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java:38]: It's a checked exception so either we wrap this call in try-catch block or add it to the method signature, that this method throws. If we add that to the method signature, then we would have to add the same throws Exception to `getUuid`, which is illegal, as it doesn't throw anything in the interface.

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java (hunk 2 above)
Comment (BartlomiejRasztabiga) [reply omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/DatabaseChangeController2_0Test.java:28]: It's a checked exception so either we wrap this call in try-catch block or add it to the method signature, that this method throws. If we add that to the method signature, then we would have to add the same throws Exception to `getUuid`, which is illegal, as it doesn't throw anything in the interface.

**Pull request merged**